sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -113,6 +113,18 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
     case object COMMITTED extends STATE
     case object ABORTED extends STATE
 
+    Option(TaskContext.get()).foreach { ctxt =>
+      ctxt.addTaskCompletionListener[Unit]( ctx => {
```

Comment:
Can we also add a note about the order in which these listeners fire and ensure that any change touching this part does not violate the underlying assumptions ?